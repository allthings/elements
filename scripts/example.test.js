import { shallow, mount, render } from 'enzyme'
import retrieveExamples from './example'
import { transform } from 'babel-core'
var { View, ThemeProvider, Spinner } = require('./atoms')
var React = require('react');

/*
describe('Ladeeda', async () => {
  const randomShit = retrieveExamples.then(result => {
    //console.log('restul result result', result)
    result.map(example => {
      const { wholeExample, file, displayName } = example
      if (displayName === 'Spinner') {
        wholeExample.forEach(stringExample => {
          console.log('working working', stringExample)
          console.log('spinner3', Spinner)
          const transformedString = transform(stringExample, {presets: ['react'], plugins: ['transform-react-jsx']}, function(err, result) {
            console.log('result is', result)
          })
          //console.log('resulttwo is', transformedString)
          //const Spin = eval('console.log("eval spiner is");' + resultTwo.code)
          //const component = shallow(Spin)
          //console.log('prop is', Spin)
        })
      }
    })
    return('jello')
  }).catch(err => {
    console.log(err)
    return('mistake')
  })
  randomShit.then(result => console.log('awaited', result))
  it('should match', async () => {
    //const examples = await retrieveExamples
    
  })
})
*/


describe('Ladeeda', async () => {
  it('should match', async () => {
    const examples = await retrieveExamples
    examples.map(example => {
      const { wholeExample, file, displayName } = example
      //console.log('example is', displayName)
      if (displayName === 'Spinner') {
        wholeExample.forEach((stringExample, index) => {
          const result = transform(stringExample, {presets: ['react']})
          console.log('result.code', result.code)
          const Spin = eval(result.code)
          const component = shallow(Spin)
          console.log('prop is', component)
          expect(component).toMatchSnapshot(displayName + index)
        })
      }
    })
  })
})




      /*
      if (wholeExample !== 'false' && wholeExample.length > 2) {
        wholeExample.forEach(ex => {
          const result = transform(ex, {presets: ['react']})
          console.log('lolilol', eval(result.code))
        })
      }
      */