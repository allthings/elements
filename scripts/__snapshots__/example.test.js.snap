// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ladeeda should match: Spinner0 1`] = `
<svg
  height="30px"
  preserveAspectRatio="xMidYMid"
  viewBox="0 0 100 100"
  width="30px"
  xmlns="http://www.w3.org/2000/svg"
>
  <rect
    fill="none"
    height="100"
    width="100"
    x="0"
    y="0"
  />
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="40"
    stroke="#ffffff"
    strokeLinecap="round"
    strokeWidth="10"
  />
  <circle
    cx="50"
    cy="50"
    fill="none"
    r="40"
    stroke="#369980"
    strokeLinecap="round"
    strokeWidth="8"
  >
    <animate
      attributeName="stroke-dashoffset"
      dur="2s"
      from="0"
      repeatCount="indefinite"
      to="502"
    />
    <animate
      attributeName="stroke-dasharray"
      dur="2s"
      repeatCount="indefinite"
      values="125.5 125.5;1 250;125.5 125.5"
    />
  </circle>
</svg>
`;
